lib/driver.rb

```diff
@@ -23,25 +24,89 @@ def initialize(input)
 
     def average_rating
       total_ratings = 0
+      trips_in_progess = 0
       @trips.each do |trip|
-        total_ratings += trip.rating
+        if trip.end_time != nil
+          total_ratings += trip.rating
+        else
+          trips_in_progess += 1
+        end
       end
 
+      trips_with_ratings = trips.length - trips_in_progess
+
       if trips.length == 0
         average = 0
       else
-        average = (total_ratings.to_f) / trips.length
+          average = (total_ratings.to_f) / trips_with_ratings
       end
 
       return average
+
     end
 
     def add_trip(trip)
       if trip.class != Trip
         raise ArgumentError.new("Can only add trip instance to trip collection")
       end
-
       @trips << trip
     end
-  end
-end
+
+    def calculate_total_revenue
+      total_revenue = 0
+      fee = 1.65
+      driver_take_home = 0.8
+      @trips.each do |trip|
+        if trip.end_time != nil
+          subtotal = trip.cost - fee
+          subtotal *= driver_take_home
+          total_revenue += subtotal
+        end
+      end
+      return total_revenue.to_f.round(2)
+    end
+
+    def calculate_total_trips_time_in_hours
+      return 0 if trips.length == 0
+
+      trip_time_lengths = []
+      @trips.each do |trip|
+        if trip.end_time != nil
+          trip_duration = trip.end_time.to_f - trip.start_time.to_f
```

Comment:
I see this condition `if trip.end_time != nil` in many places. One way you could DRY this up is by defining a helper method that returns a list of all the completed trips:

```ruby
def completed_trips
  return @trips.reject{ |t| t.end_time.nil? }
end

def total_revenue
  completed_trips.each do |trip|
    # ... same logic as before ...
  end
end
```